### PR TITLE
Module1

### PR DESCRIPTION
**Pull request by DmitriyBabii**

### CODE REVIEW

File: src/com/alevel/module1/Lock.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package com.alevel.module1;
+
+public class Lock {
+    private final int[] mass;
```
Comment (Fabolo12) [inline src/com/alevel/module1/Lock.java:4]: We discussed that it won't work.

File: src/com/alevel/module1/Main.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+
+import org.w3c.dom.ls.LSOutput;
+
+import java.util.Random;
+import java.util.Scanner;
+
+
+public class Main {
+    public static void main(String[] args) {
+        final Scanner in = new Scanner(System.in);
```
Comment (Fabolo12) [inline src/com/alevel/module1/Main.java:9]: Why create this variable?

File: src/com/alevel/module1/Main.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        lock.getArray()[0] = 2;
+        lock.printArray();
+    }
+
+    static int[] createArray() {
+        final Scanner in = new Scanner(System.in);
+        do {
+            try {
+                System.out.print("Enter the number of elements in the array from 10: ");
+                int length = Integer.parseInt(in.next());
```
Comment (Fabolo12) [inline src/com/alevel/module1/Main.java:50]: You can use in.nextInt()

File: src/com/alevel/module1/Main.java [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+            mass[i] = -range + rand.nextInt(range * 2 + 1);
+        }
+    }
+
+    static void outputArray(int[] mass) {
+        for (int i : mass) {
+            System.out.printf("%8d", i);
+        }
+        System.out.println();
+    }
```
Comment (Fabolo12) [inline src/com/alevel/module1/Main.java:73]: You can use Arrays.toString

File: src/com/alevel/module1/Main.java [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+
+    static void reverse(int[] mass) {
+        int size = mass.length;
+        for (int i = 0; i < size / 2; i++) {
+            int tmp = mass[i];
+            mass[i] = mass[size - 1 - i];
+            mass[size - 1 - i] = tmp;
+        }
+    }
```
Comment (Fabolo12) [inline src/com/alevel/module1/Main.java:103]: Interesting solution

File: src/com/alevel/module1/Main.java [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+            }
+        }
+    }
+
+    static int pow(int value, int degree) {
+        if (degree == 0) {
+            return 1;
+        }
+        return value * pow(value, degree - 1);
+    }
```
Comment (Fabolo12) [inline src/com/alevel/module1/Main.java:118]: Why is recursion here? You can use Math.pow()

**Review by Fabolo12 — APPROVED** [review]
The last task has not been completed yet.

The code is split into methods, which is fine, but not classes. Superfluous methods are written that complicate reading.

Now 8, if you finish the Lock - it will be 9

File: src/com/alevel/module1/Lock.java [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+        return (mass == null) ? null : mass.clone();
+    }
+
+    public void printArray() {
+        for (int i : mass) {
+            System.out.printf("%8d", i);
+        }
+        System.out.println();
+    }
+}
```
Comment (Fabolo12) [inline src/com/alevel/module1/Lock.java]: There is a way how to change the array with the current implementation

File: src/com/alevel/module1/Main.java (hunk 2 above)
Comment (DmitriyBabii) [reply src/com/alevel/module1/Main.java:9]: Because I am using this variable on line 20.

File: src/com/alevel/module1/Main.java (hunk 3 above)
Comment (DmitriyBabii) [reply src/com/alevel/module1/Main.java:50]: In this case, if the input is incorrect, the flow is clogged. This does not happen with the "next" method.

File: src/com/alevel/module1/Lock.java (hunk 7 above)
Comment (DmitriyBabii) [reply src/com/alevel/module1/Lock.java]: `System.out.println("Lock array:");
        Lock lock = new Lock(mass);
        lock.getArray()[0] = 2;
        lock.printArray();`

These lines from the program demonstrate the inability to change the content

Comment (Fabolo12) [reply src/com/alevel/module1/Lock.java]: This is one of two options for how you can change the array. There is one more.

File: src/com/alevel/module1/Main.java (hunk 3 above)
Comment (Fabolo12) [reply src/com/alevel/module1/Main.java:50]: You can use in.hasNextInt()

**Pull request merged**